collated/main/A0139552B.md

```diff
@@ -0,0 +1,254 @@
+                startDate == null && endDate == null) {
+            startDate = DateTime.assignStartDateToSpecifiedWeekday(recurrenceRate.timePeriod.toString());
+        }
+	}
+
+    private void initializeForEdit() {
+        taskName = null;
+        startDate = null;
+        endDate = null;
+        priority = null;
+        removeReccurence = false;
+        removeStartDate = false;
+        removeEndDate = false;
+    }
+
+    private void assignTaskNameIfPresent(Optional<String> taskNameString) {
+        if (taskNameString.isPresent() && !taskNameString.get().toString().trim().equals("")) {
+    		taskName = new Name(taskNameString.get());
+        }
+    }
+
+    private void assignStartDateIfPresent(Optional<String> startDateString) {
+        if (startDateString.isPresent()) {
+            startDate = DateTime.convertStringToDate(startDateString.get());
+            if (!DateTime.hasTimeValue(startDateString.get())) {
+                startDate = DateTime.setTimeToStartOfDay(startDate);
+            }
+        }
+    }
+
+    private void assignEndDateIfPresent(Optional<String> endDateString) {
+        if (endDateString.isPresent()) {
+            endDate = DateTime.convertStringToDate(endDateString.get());
+            if (startDate != null && !DateTime.hasDateValue(endDateString.get())) {
+                endDate = DateTime.setEndDateToStartDate(startDate, endDate);
+            }
+            if (!DateTime.hasTimeValue(endDateString.get())) {
+                endDate = DateTime.setTimeToEndOfDay(endDate);
+            }
+        }
+    }
+
+    private void assignRecurrenceRateIfPresent(Optional<String> rateString, Optional<String> timePeriodString)
+            throws IllegalValueException {
+        if (rateString.isPresent() && timePeriodString.isPresent()) {
+            recurrenceRate = new RecurrenceRate(rateString.get(), timePeriodString.get());
+        } else if (!rateString.isPresent() && timePeriodString.isPresent()) {
+            recurrenceRate = new RecurrenceRate(STRING_CONSTANT_ONE, timePeriodString.get());
+        } else if (rateString.isPresent() && !timePeriodString.isPresent()) {
+            throw new IllegalValueException(RecurrenceRate.MESSAGE_VALUE_CONSTRAINTS);
+        }
+    }
+
+    /*
+     * Assign priority depending on the level stated
+     * Otherwise leave it as null
+     */
+    private void assignPriority(Optional<String> priorityString) {
+        if (priorityString.isPresent()) {
+            switch (priorityString.get()) {
+                case ("low"): case ("l"): priority = Priority.LOW; break; 
+                case ("high"): case ("h"): priority = Priority.HIGH; break;
+                case ("medium"): case ("m"): case ("med"): priority = Priority.MEDIUM; break;
+            }
+        }
+    }
+	
+    /*
+     * Check which field is to be reset
+     */
+    private void fieldsToReset(String resetFieldString) {
+        if(resetFieldString != null){
```

Comment:
better to handle the exceptional case first and then do the main thing in the methods; 
that way you avoid arrow head kinda indentation
refer to your hand outs from one of the early lecture to see an example